Kernel/Syscalls/alarm.cpp

```diff
@@ -23,6 +23,8 @@ KResultOr<FlatPtr> Process::sys$alarm(unsigned seconds)
         // We had an existing alarm, must return a non-zero value here!
         if (previous_alarm_remaining == 0)
             previous_alarm_remaining = 1;
+
+        m_alarm_timer->clear_in_use();
```

Comment:
This should be called implicitly when either the timer handler finished executing. `cancel_timer` needs to block until the deferred timer handler is done executing (having cleared this flag), and if it can cancel it before the deferred call is executing it probably shouldn't touch it just so we can ensure that we block until the deferred call is done.

Reply:
done